### PR TITLE
Final merged dataframe containing zonal means for current month and three months prior

### PR DESCRIPTION
**Pull request by botanical**
This PR contains the construction of the final merged dataframe and uses `shift()` to include the values for the three months prior to the current month.

### CODE REVIEW

File: exploration/zonal-means.ipynb [hunk 1]
```diff
@@ -121,6 +121,63 @@
...
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "merged_precip_df = pd.DataFrame()\n",
+    "merged_precip_df[\"precip_m\"] = precip_zonal_means\n",
+    "merged_precip_df[\"precip_m_1\"] = precip_zonal_means.shift(1)\n",
+    "merged_precip_df[\"precip_m_2\"] = precip_zonal_means.shift(2)\n",
+    "merged_precip_df[\"precip_m_3\"] = precip_zonal_means.shift(3)\n",
+    "merged_precip_df"
```
Comment (chuckwondo) [inline exploration/zonal-means.ipynb:136]: This doesn't quite do the trick. Take a close look at the first few rows of this DataFrame:

```plain
                                 precip_m  precip_m_1  precip_m_2  precip_m_3
year month location_period_id                                                
2009 10    92                  288.192780         NaN         NaN         NaN
           1162                 96.912056  288.192780         NaN         NaN
           1343                 69.695549   96.912056  288.192780         NaN
           1355                105.352257   69.695549   96.912056  288.192780
           1842                239.344131  105.352257   69.695549   96.912056
```

Notice the problem? (Hint: it's _not_ the `NaN` values -- simply look at the first value and how it's shifted values align with `location_period_id` [and potentially with year/month for other values])

**Review by chuckwondo — CHANGES_REQUESTED** [review]
Since we have the results of this notebook's zonal means computations saved as csv files, let's move this work to a separate notebook. Otherwise, we're required to needlessly recompute all of the zonal means in order to construct the final dataframe. This is particularly problematic due to the flakiness of the LST zonal means computation, and even in the absence of flakiness, it still takes nearly 20 minutes to compute.

Instead, in a new notebook, simply read the csv files and construct the full dataframe from the constituent dataframes. Also, I've pushed up a few changes to the branch you created this branch from, so be sure to pull them, even though you'll be adding a new notebook instead of modifying an existing one.

File: exploration/zonal-means.ipynb (hunk 1 above)
Comment (botanical) [reply exploration/zonal-means.ipynb:136]: Ahhh I see it. Let me fix that 👀

Comment (botanical) [reply exploration/zonal-means.ipynb]: https://github.com/developmentseed/geospatial-ds-cholera-lab/pull/24/commits/323074e6f0bc9ade9221d5b58751712acf9d6b07 I updated it to group by the `location_period_id` and I believe this fixes the issue. @chuckwondo, do you mind reviewing again?

**Comment by chuckwondo** [comment]
@botanical, I suspect you missed my other comment:

> Since we have the results of this notebook's zonal means computations saved as csv files, let's move this work to a separate notebook. Otherwise, we're required to needlessly recompute all of the zonal means in order to construct the final dataframe. This is particularly problematic due to the flakiness of the LST zonal means computation, and even in the absence of flakiness, it still takes nearly 20 minutes to compute.
> 
> Instead, in a new notebook, simply read the csv files and construct the full dataframe from the constituent dataframes.

Also, I've already landed my PR on `main`, so please create a fresh branch off of `main` instead of my now-merged branch.

Finally, be careful when using `git add .`, as that opens you up for accidentally committing files you don't want to commit, such as the `.DS_Store` file that I now see as part of your PR. I tend to either add specific files, or at least run `git status` after running `git add .` to check the list of staged files, where I can then see if I've accidentally added files I didn't mean to add.

When you move this work to a fresh branch off of `main`, at a minimum, you should add `.DS_Store` to the `.gitignore` file. However, better yet, I recommend something a bit more comprehensive to cover a number of ignorable macOS files by doing the following instead (which will add `.DS_Store` to `.gitignore`, along with other files):

```plain
curl -sL https://www.toptal.com/developers/gitignore/api/macos >> .gitignore
```

For future use of this handy "ignore" functionality, I suggest creating a global git alias for yourself, so you don't have to remember the `curl` command above:

```plain
git config --global --add alias.ignore '!gi() { curl -sL https://www.toptal.com/developers/gitignore/api/$@ ;}; gi'
```

By running that command, you can then run the following command to show you all possible "types" you can ignore:

```plain
git ignore list
```

Given that "macos" is one of the valid types listed by the command above, you can then run the following command to produce a list of ignorable macOS-specific files:

```plain
git ignore macos
```

This command simply dumps output to stdout, which is handy for initially checking the list of patterns it produces that you can add to `.gitignore`. Once you confirm the output looks like the output you desire, you can then simply append the output to `.gitignore`, like so (rather than copy-pasting from the terminal to `.gitignore`):

```plain
git ignore macos >> .gitignore
```

And that command is equivalent to the `curl` command I initially gave above:

```plain
curl -sL https://www.toptal.com/developers/gitignore/api/macos >> .gitignore
```

However, by adding the git global alias, you don't have to remember the `curl` command, and you can use the alias across all of your repos on your machine.

Once you've created a new branch from `main`, submit a PR for the new branch, and close this PR. I'll then take a closer look at the contents of that new PR.

**Review by chuckwondo — CHANGES_REQUESTED** [review]
Please see my top-level comment on this PR.

**Comment by botanical** [comment]
Closing in favor of https://github.com/developmentseed/geospatial-ds-cholera-lab/pull/25

**Pull request closed without merging**